### PR TITLE
Revisit: mbedtls port: Fix detection of EWOULDBLOCK/EAGAIN with non-blocking sockets

### PR DESCRIPTION
**Pull request by mkellner**
Since mbedtls_net_errno is reset by fcntl, it is reset after calling
net_would_block, so the call to mbedtls_net_errno in mbedtls_net_recv
and mbedtls_net_send will always get back 0. This change propagates
the value returned by mbedtls_net_errno up through net_would_block,
to allow the correct error value to be used and avoid a redundant
call to mbedtls_net_errno.

### CODE REVIEW

File: components/mbedtls/port/net.c [hunk 1]
```diff
@@ -201,10 +201,12 @@ int mbedtls_net_bind( mbedtls_net_context *ctx, const char *bind_ip, const char
  *
  * Note: on a blocking socket this function always returns 0!
  */
-static int net_would_block( const mbedtls_net_context *ctx )
+static int net_would_block( const mbedtls_net_context *ctx, int *errout )
 {
     int error = mbedtls_net_errno(ctx->fd);
 
+    if ( errout ) *errout = error;
```
Comment (projectgus) [inline components/mbedtls/port/net.c]: Minor style thing, but can you please put the assignment statement on a separate line inside curly braces. This in keeping with conditional style in the rest of the file.

Comment (mkellner) [inline components/mbedtls/port/net.c]: No problem. Done.

**Comment by projectgus** [comment]
Thanks Michael. Sorry it took me so long to get back to this. I've squashed the mbedTLS commits and pushed them into our internal review & merge queue.

I've also added the SPI master driver fix d92d57cfeb.

Angus

**Comment by projectgus** [comment]
Cherry-picked as a523aa3 and eeb0aaa, thanks!

**Pull request closed without merging**